XenAdmin/Alerts/Types/ClientUpdateAlert.cs

```diff
@@ -134,17 +136,66 @@ public static void DownloadAndInstallNewClient(ClientUpdateAlert updateAlert, IW
             {
                 Process.Start(outputPathAndFileName);
                 log.DebugFormat("Update {0} found and install started", updateAlert.Name);
-                downloadAndInstallClientAction.ReleaseInstaller();
+                downloadAndInstallClientAction.ReleaseDownloadedContent();
                 Application.Exit();
             }
             catch (Exception e)
             {
                 log.Error("Exception occurred when starting the installation process.", e);
-                downloadAndInstallClientAction.ReleaseInstaller(true);
+                downloadAndInstallClientAction.ReleaseDownloadedContent(true);
 
                 using (var dlg = new ErrorDialog(Messages.UPDATE_CLIENT_FAILED_INSTALLER_LAUNCH))
                     dlg.ShowDialog(parent);
             }
         }
+
+        public static void DownloadSource(IWin32Window parent)
+        {
+            var outputPathAndFileName = string.Empty;

```

Comment:
super nit picky but we don't use `string.Empty`:


```suggestion
            string outputPathAndFileName;
```